velox/dwio/dwrf/test/QatZstdCompressionBenchmark.cpp

```diff
@@ -0,0 +1,135 @@
+#include "velox/dwio/dwrf/common/Compression.h"
+#include "velox/dwio/dwrf/common/wrap/dwrf-proto-wrapper.h"
+#include "velox/dwio/dwrf/test/OrcTest.h"
+
+#include <folly/Benchmark.h>
+#include <folly/Random.h>
+#include <folly/init/Init.h>
+#include <gtest/gtest.h>
+
+#include <algorithm>
+#include <ctime>
+#include <iostream>
+
+using namespace ::testing;
+using namespace facebook::velox::common;
+using namespace facebook::velox::dwio;
+using namespace facebook::velox::dwio::common;
+using namespace facebook::velox::dwio::common::encryption;
+using namespace facebook::velox::dwio::common::encryption::test;
+using namespace facebook::velox::dwrf;
+using namespace facebook::velox::memory;
+using namespace std;
+using namespace folly;
+
+const int32_t DEFAULT_MEM_STREAM_SIZE = 1024 * 1024 * 2; // 2M
+
+class TestBufferPool : public CompressionBufferPool {
+ public:
+  TestBufferPool(MemoryPool& pool, uint64_t blockSize)
+      : buffer_{std::make_unique<DataBuffer<char>>(
+            pool,
+            blockSize + PAGE_HEADER_SIZE)} {}
+
+  std::unique_ptr<DataBuffer<char>> getBuffer(uint64_t /* unused */) override {
+    return std::move(buffer_);
+  }
+
+  void returnBuffer(std::unique_ptr<DataBuffer<char>> buffer) override {
+    buffer_ = std::move(buffer);
+  }
+
+ private:
+  std::unique_ptr<DataBuffer<char>> buffer_;
+};
+
+void generateRandomData(char* data, size_t size, bool letter) {
+  for (size_t i = 0; i < size; ++i) {
+    if (letter) {
+      bool capitalized = folly::Random::rand32() % 2 == 0;
+      data[i] = capitalized
+          ? static_cast<char>('A' + folly::Random::rand32() % 26)
+          : static_cast<char>('a' + folly::Random::rand32() % 26);
+    } else {
+      data[i] = static_cast<char>(folly::Random::rand32() % 256);
+    }
+  }
+}
+
+void benchmarkCompress(
+    CompressionKind kind,
+    DataSink& sink,
+    uint64_t block,
+    MemoryPool& pool,
+    const char* data,
+    size_t dataSize,
+    const Encrypter* encrypter) {
+  TestBufferPool bufferPool(pool, block);
+  DataBufferHolder holder{
+      pool, block, 0, DEFAULT_PAGE_GROW_RATIO, std::addressof(sink)};
+  Config config;
+  config.set<uint32_t>(Config::COMPRESSION_THRESHOLD, 128);
+  std::unique_ptr<BufferedOutputStream> compressStream =
+      createCompressor(kind, bufferPool, holder, config, encrypter);
+  size_t pos = 0;
+  char* compressBuffer;
+  int32_t compressBufferSize = 0;
+  while (dataSize > 0 &&
+         compressStream->Next(
+             reinterpret_cast<void**>(&compressBuffer), &compressBufferSize)) {
+    size_t copy_size =
+        std::min(static_cast<size_t>(compressBufferSize), dataSize);
+    memcpy(compressBuffer, data + pos, copy_size);
+
+    if (copy_size == dataSize) {
+      compressStream->BackUp(
+          compressBufferSize - static_cast<int32_t>(dataSize));
+    }
+
+    pos += copy_size;
+    dataSize -= copy_size;
+  }
+  compressStream->flush();
+}
+
+BENCHMARK(compressZstd) {
```

Comment:
As I pointed out earlier, having a flag allowing us to switch behaviour will allow us to compare QAT inside same binary. Can we do that ?